crates/cli/src/lib.rs

```diff
@@ -73,13 +84,37 @@ fn setup_caching(mode: &CacheMode) {
     }
 }
 
+async fn check_version() -> Result<Option<String>, Box<dyn Error + Send + Sync>> {
+    let local_version_str = env!("CARGO_PKG_VERSION");
+    debug!("Checking for new version of moon");
+    let response = reqwest::Client::new()
```

Comment:
should we cache this request and only fire it like every hour/day? doing it every execution seems like too much